### PR TITLE
fix: Boxshadow displays abnormally in fractional scaling scenarios

### PR DESCRIPTION
**Pull request by lbwtw**
- Temporary solution

### CODE REVIEW

File: src/private/dquickborderimage.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+DQuickBorderImage::DQuickBorderImage(QQuickItem *parent)
+    :QQuickBorderImage(parent)
+{
+}
+
+void DQuickBorderImage::itemChange(ItemChange change, const ItemChangeData &value)
+{
+    if (change == ItemDevicePixelRatioHasChanged &&
+        value.realValue != static_cast<QQuickBorderImagePrivate *>(QQuickItemPrivate::get(this))->devicePixelRatio) {
```
Comment (zccrs) [inline src/private/dquickborderimage.cpp:20]: 只需要判断 change 就行

File: src/private/dquickborderimage.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#include "dquickborderimage.h"
+
+#include <private/qquickitem_p.h>
+#include <private/qquickborderimage_p_p.h>
+
+DQUICK_BEGIN_NAMESPACE
+
+DQuickBorderImage::DQuickBorderImage(QQuickItem *parent)
+    :QQuickBorderImage(parent)
```
Comment (zccrs) [inline src/private/dquickborderimage.cpp:13]: `: ` 别忘记了空格

File: src/private/dquickborderimage.h
Comment (zccrs) [inline src/private/dquickborderimage.h:1]: _p.h

File: src/private/dquickborderimage.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+DQUICK_BEGIN_NAMESPACE
+
+class DQuickBorderImage : public QQuickBorderImage
+{
+    Q_OBJECT
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    QML_NAMED_ELEMENT(DBorderImage)
+#endif
+public:
+    DQuickBorderImage(QQuickItem *parent=nullptr);
```
Comment (zccrs) [inline src/private/dquickborderimage.h:22]: explicit

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**